ci: test-images: intermittent failure

https://github.com/getodk/central/actions/runs/20834960813/job/59857785581

# Logs

## `./test/test-images.sh`

```
 Container central-nginx-1  Started
[test-images.sh] Verifying frontend...
[test-images.sh] Checking response from / ...
[test-images.sh]   Request did not respond correctly; sleeping...
[test-images.sh] Checking response from / ...
[test-images.sh]   Request did not respond correctly; sleeping...
[test-images.sh] Checking response from / ...
[test-images.sh]   Request responded correctly.
[test-images.sh]   Frontend started OK.
[test-images.sh] Verifying backend...
[test-images.sh] Checking response from /v1/projects ...
[test-images.sh]   Request did not respond correctly; sleeping...
[test-images.sh] Checking response from /v1/projects ...
[test-images.sh]   Request did not respond correctly; sleeping...
[test-images.sh] Checking response from /v1/projects ...
[test-images.sh] !!! Retry count exceeded.
[test-images.sh] !!! Final response:

Connecting to 127.0.0.1
Can't use SSL_get_servername
depth=0 CN=invalid.local
verify error:num=18:self-signed certificate
verify return:1
depth=0 CN=invalid.local
verify return:1
HTTP/1.1 502 Bad Gateway
Server: nginx
Date: Thu, 08 Jan 2026 23:12:52 GMT
Content-Type: text/html
Content-Length: 150
Connection: close
Strict-Transport-Security: max-age=63072000

<html>
<head><title>502 Bad Gateway</title></head>
<body>
<center><h1>502 Bad Gateway</h1></center>
<hr><center>nginx</center>
</body>
</html>

##[error]Process completed with exit code 1.
```

## `docker compose logs`

```
�[36;1mdocker compose logs�[0m
shell: /usr/bin/bash -e {0}
##[endgroup]
service-1  | wait-for-it: waiting 15 seconds for postgres14:5432
service-1  | wait-for-it: postgres14:5432 is available after 3 seconds
service-1  | generating local service configuration..
service-1  | running migrations..
service-1  | Migrating timestamps, this may take a while if you have a lot of submissions.
service-1  | *** This migration [20230109-01-add-form-schema.js] may take a long time ***
service-1  | The purpose of this migration is to clean up redundant data related to how form schemas are
service-1  | stored. Please be aware that it may take a while to complete if there are many different
service-1  | versions of forms and/or many large forms with many fields in the database.
service-1  | logging server upgrade..
service-1  | null
service-1  | starting cron..
service-1  | using 4 worker(s) based on available memory (16772600000)..
service-1  | starting server.
enketo_redis_main-1  | 1:C 08 Jan 2026 23:12:46.624 # WARNING Memory overcommit must be enabled! Without it, a background save or replication may fail under low memory condition. Being disabled, it can also cause failures without low memory condition, see https://github.com/jemalloc/jemalloc/issues/1328. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
enketo_redis_main-1  | 1:C 08 Jan 2026 23:12:46.624 * oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
enketo_redis_main-1  | 1:C 08 Jan 2026 23:12:46.624 * Redis version=7.4.7, bits=64, commit=00000000, modified=0, pid=1, just started
enketo_redis_main-1  | 1:C 08 Jan 2026 23:12:46.624 * Configuration loaded
enketo_redis_main-1  | 1:M 08 Jan 2026 23:12:46.625 * monotonic clock: POSIX clock_gettime
enketo_redis_main-1  | 1:M 08 Jan 2026 23:12:46.626 * Running mode=standalone, port=6379.
enketo_redis_main-1  | 1:M 08 Jan 2026 23:12:46.626 * Server initialized
enketo_redis_main-1  | 1:M 08 Jan 2026 23:12:46.627 * Ready to accept connections tcp
postgres-1           | Thu Jan  8 11:12:46 PM GMT 2026 [upgrade-postgres.sh] Checking for existing upgrade marker file...
postgres-1           | Thu Jan  8 11:12:46 PM GMT 2026 [upgrade-postgres.sh] No old data found.
postgres-1           | Thu Jan  8 11:12:46 PM GMT 2026 [upgrade-postgres.sh] Complete.
pyxform-1            | [2026-01-08 23:12:46 +0000] [1] [INFO] Starting gunicorn 23.0.0
pyxform-1            | [2026-01-08 23:12:46 +0000] [1] [INFO] Listening at: http://0.0.0.0:80 (1)
pyxform-1            | [2026-01-08 23:12:46 +0000] [1] [INFO] Using worker: sync
pyxform-1            | [2026-01-08 23:12:46 +0000] [6] [INFO] Booting worker with pid: 6
pyxform-1            | [2026-01-08 23:12:46 +0000] [7] [INFO] Booting worker with pid: 7
pyxform-1            | [2026-01-08 23:12:46 +0000] [8] [INFO] Booting worker with pid: 8
pyxform-1            | [2026-01-08 23:12:46 +0000] [9] [INFO] Booting worker with pid: 9
pyxform-1            | [2026-01-08 23:12:46 +0000] [10] [INFO] Booting worker with pid: 10
mail-1               | using non-split configuration scheme from /etc/exim4/exim4.conf.template
mail-1               |     1 LOG: MAIN
mail-1               |     1   exim 4.98.2 daemon started: pid=1, -q15m, listening for SMTP on port 25 (IPv6 and IPv4)
mail-1               |   273 Exim version 4.98.2 uid=100 gid=101 pid=273 D=80001
mail-1               |   273 Hints DB:
mail-1               |   273  Berkeley DB: Berkeley DB 5.3.28: (September  9, 2013)
mail-1               |   273 Support for: crypteq iconv() IPv6 GnuTLS move_frozen_messages TLS_resume DANE DKIM DNSSEC ESMTP_Limits ESMTP_Wellknown Event I18N OCSP PIPECONNECT PRDR Queue_Ramp SOCKS SRS TCP_Fast_Open
mail-1               |   273 Lookups (built-in): lsearch wildlsearch nwildlsearch iplsearch cdb dbm dbmjz dbmnz dnsdb dsearch nis nis0 passwd
mail-1               |   273 Authenticators: cram_md5 external plaintext
mail-1               |   273 Routers: accept dnslookup ipliteral manualroute queryprogram redirect
mail-1               |   273 Transports: appendfile/maildir/mailstore autoreply lmtp pipe smtp
mail-1               |   273 Fixed never_users: 0
mail-1               |   273 Configure owner: 0:0
mail-1               |   273 Size of off_t: 8
mail-1               |   273 LOG: queue_run MAIN
mail-1               |   273   Start queue run: pid=273
mail-1               |   273 LOG: queue_run MAIN
mail-1               |   273   End queue run: pid=273
enketo_redis_cache-1  | 1:C 08 Jan 2026 23:12:46.566 # WARNING Memory overcommit must be enabled! Without it, a background save or replication may fail under low memory condition. Being disabled, it can also cause failures without low memory condition, see https://github.com/jemalloc/jemalloc/issues/1328. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
enketo_redis_cache-1  | 1:C 08 Jan 2026 23:12:46.566 * oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
enketo_redis_cache-1  | 1:C 08 Jan 2026 23:12:46.566 * Redis version=7.4.7, bits=64, commit=00000000, modified=0, pid=1, just started
enketo_redis_cache-1  | 1:C 08 Jan 2026 23:12:46.566 * Configuration loaded
enketo_redis_cache-1  | 1:M 08 Jan 2026 23:12:46.566 * monotonic clock: POSIX clock_gettime
enketo_redis_cache-1  | 1:M 08 Jan 2026 23:12:46.567 * Running mode=standalone, port=6380.
enketo_redis_cache-1  | 1:M 08 Jan 2026 23:12:46.567 * Server initialized
enketo_redis_cache-1  | 1:M 08 Jan 2026 23:12:46.567 * Ready to accept connections tcp
postgres14-1          | Thu Jan  8 11:12:46 PM GMT 2026 [start-postgres.sh] Waiting for upgrade to complete...
postgres14-1          | Thu Jan  8 11:12:47 PM GMT 2026 [start-postgres.sh] Upgrade complete.
postgres14-1          | Thu Jan  8 11:12:47 PM GMT 2026 [start-postgres.sh] Starting postgres...
postgres14-1          | The files belonging to this database system will be owned by user "postgres".
postgres14-1          | This user must also own the server process.
postgres14-1          | 
postgres14-1          | The database cluster will be initialized with locale "en_US.utf8".
postgres14-1          | The default database encoding has accordingly been set to "UTF8".
postgres14-1          | The default text search configuration will be set to "english".
postgres14-1          | 
postgres14-1          | Data page checksums are disabled.
postgres14-1          | 
postgres14-1          | fixing permissions on existing directory /var/lib/odk/postgresql/14/data ... ok
postgres14-1          | creating subdirectories ... ok
postgres14-1          | selecting dynamic shared memory implementation ... posix
postgres14-1          | selecting default max_connections ... 100
postgres14-1          | selecting default shared_buffers ... 128MB
postgres14-1          | selecting default time zone ... Etc/UTC
postgres14-1          | creating configuration files ... ok
postgres14-1          | running bootstrap script ... ok
postgres14-1          | performing post-bootstrap initialization ... ok
postgres14-1          | syncing data to disk ... ok
postgres14-1          | 
postgres14-1          | 
postgres14-1          | Success. You can now start the database server using:
postgres14-1          | 
postgres14-1          |     pg_ctl -D /var/lib/odk/postgresql/14/data -l logfile start
postgres14-1          | 
postgres14-1          | initdb: warning: enabling "trust" authentication for local connections
postgres14-1          | You can change this by editing pg_hba.conf or using the option -A, or
postgres14-1          | --auth-local and --auth-host, the next time you run initdb.
postgres14-1          | waiting for server to start....2026-01-08 23:12:49.285 UTC [53] LOG:  starting PostgreSQL 14.20 (Debian 14.20-1.pgdg13+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 14.2.0-19) 14.2.0, 64-bit
postgres14-1          | 2026-01-08 23:12:49.286 UTC [53] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres14-1          | 2026-01-08 23:12:49.295 UTC [54] LOG:  database system was shut down at 2026-01-08 23:12:49 UTC
postgres14-1          | 2026-01-08 23:12:49.306 UTC [53] LOG:  database system is ready to accept connections
postgres14-1          |  done
postgres14-1          | server started
postgres14-1          | CREATE DATABASE
postgres14-1          | 
postgres14-1          | 
postgres14-1          | /usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/*
postgres14-1          | 
postgres14-1          | 2026-01-08 23:12:49.593 UTC [53] LOG:  received fast shutdown request
postgres14-1          | waiting for server to shut down....2026-01-08 23:12:49.596 UTC [53] LOG:  aborting any active transactions
postgres14-1          | 2026-01-08 23:12:49.602 UTC [53] LOG:  background worker "logical replication launcher" (PID 60) exited with exit code 1
postgres14-1          | 2026-01-08 23:12:49.603 UTC [55] LOG:  shutting down
postgres14-1          | 2026-01-08 23:12:49.622 UTC [53] LOG:  database system is shut down
postgres14-1          |  done
postgres14-1          | server stopped
postgres14-1          | 
postgres14-1          | PostgreSQL init process complete; ready for start up.
postgres14-1          | 
postgres14-1          | 2026-01-08 23:12:49.751 UTC [1] LOG:  starting PostgreSQL 14.20 (Debian 14.20-1.pgdg13+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 14.2.0-19) 14.2.0, 64-bit
postgres14-1          | 2026-01-08 23:12:49.751 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgres14-1          | 2026-01-08 23:12:49.751 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgres14-1          | 2026-01-08 23:12:49.752 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres14-1          | 2026-01-08 23:12:49.767 UTC [68] LOG:  database system was shut down at 2026-01-08 23:12:49 UTC
postgres14-1          | 2026-01-08 23:12:49.774 UTC [1] LOG:  database system is ready to accept connections
postgres14-1          | 2026-01-08 23:12:50.145 UTC [75] LOG:  incomplete startup packet
nginx-1               | writing client config...
nginx-1               | ........+..+....+...+.................+.+++++++++++++++++++++++++++++++++++++++*............+....+..+...............+....+........+...+.......+.....+...+++++++++++++++++++++++++++++++++++++++*.......+......+..+....+...+..+..........+......+..+...+....+......+....................+...+......+......+...+.+.........+..+.......+..+...............+......+....+......+...+......+.....+...............+....+...........+.+........+....+...+........+....+......+.........+.....+...+...+....+......+..+....+......+...+...........+.+..+.+.....+....+...+.....+....+.....+....+..+..........+...+......+...........+...+....+......+.....+......+.+..+.............+..+......+.....................+....+..............+...+............+.......+......+..+.......+...............+.....+...+.+......+..............+....+..+.........+....+...+............+.......................+....++++++
nginx-1               | .........+.+..+++++++++++++++++++++++++++++++++++++++*.......+...+++++++++++++++++++++++++++++++++++++++*...+...+.......+...........+...............+...............+......+.+.....+...+...............+....+.................+...................+..+..........+........+.+...............+........+.+...........+......+.......+..+....+.....+............+...............+...+.+...+...+..+...+....+...+..+.+..+......+......+.............+..+.+........+....+...........++++++
nginx-1               | -----
nginx-1               | Generating DH parameters, 2048 bit long safe prime
nginx-1               | ....................................................................................................................................................................................................................................................................+...........................................................................................................................................+.................................+.....++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*++*
nginx-1               | ..+...+......+.........+..+....+......+..+.+..+....++++++++++++++++++++++++++++++++++++++++++*......++++++++++++++++++++++++++++++++++++++++++*...........+.+...+......+......+.........+...+...+.......................+....+.........................................+......+.+..+.......+....................+........................+.......+..................+..+.......+...........+....+............+......+.....+............+...+...........................+..........+......+......+...+...........+....+..+.+..+......+...............+.......+..+............+....+..+........................+......+............+..........+...+++++
nginx-1               | ..+...+......+..++++++++++++++++++++++++++++++++++++++++++*..+...........+......+....+..+....+...+...+.....+...++++++++++++++++++++++++++++++++++++++++++*..+.+...........+....+..+.......+...............+......+.....+..............................+....+...+..+..........+...+...+.....+......+.+............+..+.+..............+.+...+..+...+......+..........+++++
nginx-1               | -----
nginx-1               | writing fresh nginx templates...
nginx-1               | starting nginx for custom ssl and self-signed certs...
nginx-1               | 2026/01/08 23:12:50 [notice] 1#1: using the "epoll" event method
nginx-1               | 2026/01/08 23:12:50 [notice] 1#1: nginx/1.29.4
nginx-1               | 2026/01/08 23:12:50 [notice] 1#1: built by gcc 14.2.0 (Debian 14.2.0-19) 
nginx-1               | 2026/01/08 23:12:50 [notice] 1#1: OS: Linux 6.11.0-1018-azure
nginx-1               | 2026/01/08 23:12:50 [notice] 1#1: getrlimit(RLIMIT_NOFILE): 65536:65536
nginx-1               | 2026/01/08 23:12:50 [notice] 1#1: start worker processes
nginx-1               | 2026/01/08 23:12:50 [notice] 1#1: start worker process 29
nginx-1               | 2026/01/08 23:12:50 [notice] 1#1: start worker process 30
nginx-1               | 2026/01/08 23:12:50 [notice] 1#1: start worker process 31
nginx-1               | 2026/01/08 23:12:50 [notice] 1#1: start worker process 32
nginx-1               | 127.0.0.1 - - [08/Jan/2026:23:12:50 +0000] "GET / HTTP/1.0" 200 1116 "-" "-" "-"
nginx-1               | 2026/01/08 23:12:50 [error] 31#31: *2 connect() failed (111: Connection refused) while connecting to upstream, client: 127.0.0.1, server: local, request: "GET /v1/projects HTTP/1.0", upstream: "http://172.18.0.7:8383/v1/projects", host: "local"
nginx-1               | 127.0.0.1 - - [08/Jan/2026:23:12:50 +0000] "GET /v1/projects HTTP/1.0" 502 150 "-" "-" "-"
nginx-1               | 2026/01/08 23:12:51 [error] 30#30: *4 connect() failed (111: Connection refused) while connecting to upstream, client: 127.0.0.1, server: local, request: "GET /v1/projects HTTP/1.0", upstream: "http://172.18.0.7:8383/v1/projects", host: "local"
nginx-1               | 127.0.0.1 - - [08/Jan/2026:23:12:51 +0000] "GET /v1/projects HTTP/1.0" 502 150 "-" "-" "-"
nginx-1               | 2026/01/08 23:12:52 [error] 32#32: *6 connect() failed (111: Connection refused) while connecting to upstream, client: 127.0.0.1, server: local, request: "GET /v1/projects HTTP/1.0", upstream: "http://172.18.0.7:8383/v1/projects", host: "local"
nginx-1               | 127.0.0.1 - - [08/Jan/2026:23:12:52 +0000] "GET /v1/projects HTTP/1.0" 502 150 "-" "-" "-"
enketo-1              | [start-enketo.sh] Checking secrets exist...
enketo-1              | [start-enketo.sh] Generating enketo configuration...
enketo-1              | [start-enketo.sh] Starting enketo...
enketo-1              | yarn workspace v1.22.22
enketo-1              | yarn run v1.22.22
enketo-1              | $ node .
enketo-1              | Worker 1 ready for duty at port 8005! (environment: production)
enketo-1              | Worker 4 ready for duty at port 8005! (environment: production)
enketo-1              | Worker 2 ready for duty at port 8005! (environment: production)
enketo-1              | Worker 3 ready for duty at port 8005! (environment: production)
```

--- a/test/test-images.sh
+++ b/test/test-images.sh
@@ -54,7 +54,7 @@ check_path 180 / 'ODK Central'
 log "  Frontend started OK."
 
 log "Verifying backend..."
-check_path 2 /v1/projects '{"message":"Could not authenticate with the provided credentials.","code":401.2}'
+check_path 20 /v1/projects '{"message":"Could not authenticate with the provided credentials.","code":401.2}'
 log "  Backend started OK."
 
 log "Verifying enketo..."

